File: BOJ 2/10828.cpp
#include <stdio.h>
#include <string.h>
#include <algorithm>
#include <iostream>
#include <vector>
#include <queue>
#include <stack>
#include <map>
#include <deque>
using namespace std;
int arr[10005];
int n,ptr=-1;
void push(int num) {
	arr[++ptr] = num;
}
int pop() {
	if (ptr < 0) return -1;
	else {
		return arr[ptr--];
	}
}
int main() {
	scanf("%d", &n);
	for (int i = 1; i <= n; i++) {
		char s[15] = { '\0', };
		scanf("%s", s);
		if (!strcmp(s, "push")) {
			int num;
			scanf("%d", &num);
			push(num);
		}
		else if (!strcmp(s, "pop")) {
			printf("%d\n", pop());
		}
		else if (!strcmp(s, "size")) {
			printf("%d\n", ptr + 1);
		}
		else if (!strcmp(s, "empty")) {
			if (ptr < 0) printf("1\n");
			else printf("0\n");
		}
		else if (!strcmp(s, "top")) {
			if (ptr < 0) printf("-1\n");
			else printf("%d\n", arr[ptr]);
		}
	}
	return 0;
}

File: BOJ(Python 공부용)/10828.py
#!/usr/bin/env python

import sys
import collections
stack = collections.deque()
n = input()
n = int(n)
for i in range(n):
    choice = sys.stdin.readline().strip()
    if choice[:4] == "push":
        num = int(choice.split(' ')[1])
        stack.append(num)
    if choice[:3] == "pop":
        if not stack:
            print("-1")
            continue
        print("%d" %stack.pop())
    if choice[:4] == "size":
        print("%d" %len(stack))
    if choice[:5] == "empty":
        if stack:
            print(0)
            continue
        print(1)
    if choice[:3] == "top":
        if not stack:
            print("-1")
            continue
        print("%d" %stack[-1])

